Optimizations

harishbommakanti opened the pull request:
Switching from default html pages to a jekyll based setup where include statements are used to simplify code and reduce possibility of buggy code

harishbommakanti committed:
Create CNAME file

harishbommakanti committed:
Merge branch 'master' into gh-pages

harishbommakanti committed:
Minor change in the nav list html code

- home.html --> index.html

harishbommakanti committed:
Merge branch 'master' into gh-pages

harishbommakanti committed:
Merge branch 'master' into gh-pages

harishbommakanti committed:
Merge branch 'master' into gh-pages

harishbommakanti committed:
Initial Commit: Set up Jekyll environment

- installed rubyinstaller/ruby/gem/jekyll and set up all the jekyll environment

harishbommakanti committed:
Created gitignore, removed jekyll-cache and _site/ files

- gitignore updated to ignore cache and site output files
- removed cache and site output files

harishbommakanti committed:
Created nav.html file

- added nav.html to file, will start to implement it in the pages

harishbommakanti committed:
Used jekyll include for nav button

-  added nav button html code to nav.html and used liquid include tags to reference it

harishbommakanti committed:
Updated nav.html: added liquid comment

harishbommakanti committed:
Created head.html

- copied all of the css/js/font sheets to an includes head.html file

harishbommakanti committed:
Used jekyll include head.html on all pages

- used jekyll include head.html on all pages
- fixed a minor bug for button styling

harishbommakanti committed:
Created a basic layout file/directory

harishbommakanti committed:
Added code to the basic.html layout

- has the template for the basic layout on all pages
- modified season page to make its body tag similar to others (instead of onchange, I just added that to the doc.onReady in the js file)

harishbommakanti committed:
Used basic.html as a layout on all pages

- used basic.html as a layout on all html pages to avoid copying over code ie increasing possibility of buggy code

harishbommakanti committed:
minor fix: deleting commented out code

- deleted old commented out html code

mshafer1 commented on CNAME:
```diff
@@ -0,0 +1 @@
+scouting.rouserobotics.com
```
didn't this already exist??

mshafer1 replied:
oh, it's just in `gh-pages`. It should be in `master` too. This is fine.

mshafer1 commented on scripts/seasonConfigScript.js:
```diff
@@ -205,6 +193,7 @@ function season_config_cookie_name(season,element)
 
 $(document).ready(function()
 {
+    loadSeasons();
```
👍  - moving the call to be with the rest of the JS code.

mshafer1 reviewed (approved):
Looks good!

mshafer1 commented on .gitignore:
```diff
@@ -0,0 +1,2 @@
+.jekyll-cache
```
put a slash on the end of this since its a folder

harishbommakanti committed:
PR fix: updated gitignore file

will change the navbar to be better on mobile devices

harishbommakanti committed:
Added css/html for a navbar

- used w3css and my own styling to go from a navbar button to an actual navbar
- deleted navbar button code (js/html/css)
- will now work on creating an active tab

harishbommakanti committed:
Added more styling to navbar

mshafer1 commented on index.html:
```diff
@@ -2,17 +2,16 @@
-        <br><br>● Go to the <a href="importing-exporting-data.html">Importing and Exporting Data</a> page to import/export currently stored data in the Season Configuration
+        <br><br>● Go to the Event Data page to enter events and teams attending that event.
+        <br><br>● Go to the Team Data page to enter and view team data such as pit scouting and match info for a given event.
+        <br><br>● Go to the Importing and Exporting Data page to import/export currently stored data in the Season Configuration
```
If you're going to have the text, go ahead and keep the links.
Friend of mine once said:

> Use-ability basically boils down to having multiple ways to do something.

What he meant was that if I'm scrolling down through this and reading the instructions, I'm not going to want to go back to the menu to find the link I need.

mshafer1 commented on w3css.css:
```diff
@@ -37,7 +37,7 @@ hr{border:0;border-top:1px solid #eee;margin:20px 0}
 .w3-table td,.w3-table th,.w3-table-all td,.w3-table-all th{padding:8px 8px;display:table-cell;text-align:left;vertical-align:top}
 .w3-table th:first-child,.w3-table td:first-child,.w3-table-all th:first-child,.w3-table-all td:first-child{padding-left:16px}
-.w3-btn,.w3-button{border:none;display:inline-block;padding:8px 16px;vertical-align:middle;overflow:hidden;text-decoration:none;color:inherit;background-color:inherit;text-align:center;cursor:pointer;white-space:nowrap}
+.w3-btn,.w3-button{border:none;display:inline-block;padding:8px 16px;vertical-align:middle;overflow:hidden;text-decoration:none;text-align:center;cursor:pointer;white-space:nowrap}
```
why is this file changing? You should pretty much just have a release version checked in . . .

harishbommakanti replied:
I was just experimenting with the navbar styling but I don't think major things have changed. I'll make sure not to modify the file from now on

mshafer1 commented on _includes/nav.html:
```diff
@@ -1,13 +1,11 @@
-<div>
-    <h1 id = "title">FRC 6321 Scouting</h1>
+
+<div class = 'navbar w3-bar' style='text-align: center'>
```
might want to check out the collapsing navbar: https://www.w3schools.com/w3css/tryit.asp?filename=tryw3css_navbar_hide

One way to do this easily, is to put the pages you want to show in the menu in a data file (in Jekyll, that means a yaml file under `_data` - I did this on [my blog](https://github.com/mshafer1/website/blob/master/_data/pages.yaml)

Your `_data/pages.yaml` file might look like:
```yaml
- title: Season Configuration
  url: /season-config.html
- title: Event Data
  url: /event-data.html
- title: Team Data
  url: /team-data.html
- title: Import/Export Data
  url: /importing-exporting-data.html
```

Then, you can have Jekyll build up the two different lists (I have an example of that [here](https://github.com/mshafer1/quizgen_music_theory/blob/master/_includes/nav.html))

Might look like:
```html
{% comment %} tag for this page is {% include nav.html %} {% endcomment %}

{% capture page_list %}
{% for page in site.data.pages %}
	<a href="{{ page.url }}" class="w3-bar-item w3-mobile w3-button">{{ page.title }}</a>
{% endfor %}
{% endcapture %}

<div class="w3-bar w3-black">
    <a href="/" class="w3-bar-item w3-button">Home</a>
    <a href="/about.html" class="w3-bar-item w3-button">About</a>
    <span class="w3-hide-small w3-right">
        {{ page_list }}
    </span>
    <a href="javascript:void(0)" class="w3-bar-item w3-button w3-right w3-hide-large w3-hide-medium" onclick="toggleNav()">&#9776;</a>
  </div>
  
  <div class="w3-bar-block w3-red w3-hide w3-hide-large w3-hide-medium js-collapse-nav">
    {{ page_list }}
  </div>

<script>
        function toggleNav() {
            $('.js-collapse-nav').toggleClass("w3-hide");
        }
</script>
```
```

harishbommakanti replied:
This was very helpful, it does make the navbar look much better on the mobile version

harishbommakanti replied:
Do you think that it might be better to remove the "FRC 6321 Scouting" title at the top of the page and make the top bar stick to the very top of the page?

mshafer1 commented on styles.css:
```diff
@@ -4,17 +4,13 @@
 }
 
-h1, h2,h3,h4, h5, h6, p,form,div,table{
+h1, h2,h3,h4, h5, h6, p,form,div,table,title{
```
You should put the `font-family` on `html` `body` (rather than listing every single type of element here)

mshafer1 reviewed (changes requested):
Personally, I prefer top bars on tall site like this (at least on mobile) - but you have to be careful to have it not become small and unusable on mobile - checkout my comments on that.

harishbommakanti committed:
PR fix: re-added links to different pages on index.html

harishbommakanti committed:
PR fix: updated styles.css for a more efficient way of making all elements ubuntu maroon color

harishbommakanti committed:
Created pages.yaml, added _data folder

- added file to make the collapsable navbar on smaller screens easier to use with jekyll templating

harishbommakanti committed:
Navbar more friendly to smaller screens

- now the navbar is collapsable on smaller screens

harishbommakanti committed:
PR fix: moved title to navbar

- moved title as part of the red navbar chunk as it looks a little better

mshafer1 commented on _layouts/basic.html:
```diff
@@ -3,9 +3,9 @@
+        <!--<div>
             <h1 id = "title">FRC 6321 Scouting</h1>
-        </div>
+        </div>-->
```
remove code and commit - don't comment out

mshafer1 commented on _includes/nav.html:
```diff
@@ -6,12 +6,15 @@
+<div style='background-color: maroon'>
+    <h1 id = "title">FRC 6321 Scouting</h1>
+    <div class = 'navbar w3-bar' style='text-align: center'>
+        <a class = 'nav-button w3-bar-block w3-button' href="/">Home</a>
```
This makes it take its own line (if this is intentional, ok, but a little unusual).

You could make it be part of the primary line with: 

```html
      <div class = 'navbar w3-bar' style='text-align: center'>
        <div id = "title" class="w3-bar-item">FRC 6321 Scouting</div>
        <a class = 'nav-button w3-bar-block w3-button' href="/">Home</a>
        <span class = 'w3-hide-small'>
...
```

mshafer1 commented on _includes/nav.html:
```diff
@@ -1,11 +1,32 @@
+        <span class = 'w3-hide-small'>
+            {{ page_list }}
+        </span>
+        <a href="javascript:void(0)" class="nav-button w3-bar-item w3-button w3-right w3-hide-large w3-hide-medium" onclick="toggleNav()">&#9776;</a>
```
To make this work properly, you need to include the following line in the `head` (via head.html)

```html
<meta name="viewport" content="width=device-width, initial-scale=1.0">
```

Docs: https://www.w3schools.com/html/html_responsive.asp

Basically, this tells the browser to adjust the size of the site to fit the screen - this should fix it looking way to small on actual mobile (but working fine when you resize Chrome down - this is caused by difference in resolution sizes).

harishbommakanti replied:
Thanks, this makes the chrome simulator work.

harishbommakanti committed:
PR fix: removed commented code in basic.html

harishbommakanti committed:
PR fix: simplified html of navbar

- moved the title into the navbar div

harishbommakanti committed:
PR fix: meta tag edited, some navbar styling

- meta tag edited so the site renders well on the chrome editor

harishbommakanti committed:
PR fix: title and menu are on same line

- added an extra title div that adjusts so its on the same line with the menu button, makes all the home/season config... buttons centered while not making the navbar look too bad

harishbommakanti committed:
Removed github/instagram/twitter links from main page

- removed html/css for social media links on front page as they are 1) not really linked to frc scouting and 2) kind of hard to style across screen sizes

harishbommakanti committed:
Further title styling, removed some uneccesary text on index.html

mshafer1 commented on _includes/head.html:
```diff
@@ -3,7 +3,7 @@
     <link rel="shortcut icon" href="images/club_logo.png" type="image/x-icon"/>
     <title>FRC 6321 Scouting</title>
-    <meta charset="UTF-8">
+    <meta charset="UTF-8" name='viewport' content="width=device-width, initial-scale=1.0">
```
seperate meta lines, first says what character mapping you're using, the second sais what to set the viewport to.

```html
<meta charset="UTF-8" />
<meta name='viewport' content="width=device-width, initial-scale=1.0" />
```

mshafer1 commented on team-data.html:
```diff
@@ -1,114 +1,81 @@
-                    The answers you save will override previous answers.
-                </h4>
-                <!--<h4 id='questionConfirmation'></h4>-->
+<div class='w3-container'>
```
You're not closing this div, and it's causing some alignment issues with the content above versus below.

Looking at the mobile simulator without it, everything looks good - so I think just remove it.

mshafer1 reviewed (approved):
looking pretty good. make these last couple of changes, and start co-ordinating with Mr. Smerek to do a demo with the team (and please loop me in).

harishbommakanti committed:
PR fix: separated meta tags

harishbommakanti committed:
PR fix: removed unecessary div tag from team-data.html
